api/Makefile

```diff
@@ -0,0 +1,66 @@
+# API Module Makefile
+# This Makefile provides targets for the swagger generation and validation
+# which are specific to the API module.
+
+DOCKER ?= docker
+BUILDX ?= $(DOCKER) buildx
+
+API_DIR := $(CURDIR)
+PROJECT_PATH := /go/src/github.com/docker/docker
+
+DOCKER_ENVS := -e GOPATH=/go
+
+DOCKER_MOUNT := -v "$(API_DIR):$(PROJECT_PATH)/api"
+
+DOCKER_IMAGE := docker-api-dev
+
+DOCKER_WORKDIR := -w $(PROJECT_PATH)/api
+
+DOCKER_FLAGS := $(DOCKER) run --rm $(DOCKER_ENVS) $(DOCKER_MOUNT) $(DOCKER_WORKDIR)
+DOCKER_RUN := $(DOCKER_FLAGS) "$(DOCKER_IMAGE)"
+
+DOCKER_CONTAINER_NAME := $(if $(CONTAINER_NAME),--name $(CONTAINER_NAME),)
+
+BUILD_CMD := $(BUILDX) build
+GO_VERSION ?= 1.25
+SWAGGER_VERSION ?= v0.31.0
+SWAGGER_DOCS_PORT ?= 9000
+
+.DEFAULT_GOAL := help
+
+.PHONY: build
+build:
+	$(BUILD_CMD) \
+		--build-arg GO_VERSION=$(GO_VERSION) \
+		--build-arg SWAGGER_VERSION=$(SWAGGER_VERSION) \
+		--target dev \
+		--load \
+		-t "$(DOCKER_IMAGE)" \
+		-f Dockerfile \
+		.
+
+.PHONY: swagger-gen
+swagger-gen: build ## generate swagger API types
+	$(DOCKER_RUN) ./scripts/generate-swagger-api.sh
+
+.PHONY: swagger-docs
+swagger-docs: ## preview the API documentation
+	@echo "API docs preview will be running at http://localhost:$(SWAGGER_DOCS_PORT)"
+	@docker run --rm -v $(PWD)/swagger.yaml:/usr/share/nginx/html/swagger.yaml \
+		-e 'REDOC_OPTIONS=hide-hostname="true" lazy-rendering' \
+		-p $(SWAGGER_DOCS_PORT):80 \
+		bfirsh/redoc:1.14.0
+
+.PHONY: validate-swagger
```

Comment:
Validation not working (intentional). Need clarification in -> https://github.com/moby/moby/pull/51300#:~:text=Currently%2C%20my%20implementation,Thanks%20in%20advance!